docker/README.md

```diff
@@ -0,0 +1,120 @@
+- `DockerFile-tave, resources folder: theia-app-vscode-ext/`: Dockerfile and resources needed to build a Trace Viewer Appliance, using a Theia application that includes the _Trace Viewer for VSCode_ _(vsix) extension_, installed [from the open vsx public registry][vsix package]
+
+Note:
+
+- the images will contain exclusively the trace viewer front-end. If you want to run a complete _trace viewer_ application, you will need a service running the trace-server (not included here)
+
+## Quickly Build/Run the example Docker images
+
+(_see section further down to learn how to manually build/run the images_)
+
+For convenience and use in CI, there are `package.json scripts` to build the docker images on Linux. Note that it's not necessary to build anything in the repo first:
+
+```bash
+$ cd <repo>/docker
+
+# build all docker images
+$ yarn build:docker
+
+
+# build docker images individually
+
+# build "tate" image: Theia app that uses 
+# the theia trace viewer extension:
+$ yarn build:docker:tate
+# Run image:
+$ yarn run:docker:tate
+
+
+# build "tave" image: Theia app that uses 
+# the vscode trace viewer extension:
+$ yarn build:docker:tave
+# Run image:
+$ yarn run:docker:tave
+
+# Access the application at http://localhost:4000/
+# e.g.:
+$ firefox http://localhost:4000
+
+```
+
+## Manually Build/run the example Docker images
+
+The Docker images can be build from the `<repo>/docker` folder. For each image, the resources it needs are in a sub-folder, whose name is passed as a `docker build` build argument and used to retrieve expected resourced needed during image build.
+
+### Build/Run the "tate" image
+
+Build the image and name it _tv-tate_ (short for `"trace viewer - theia application/theia extension"`). You may use a different name if you prefer.
+
+Try adding `--network host` in case of build failures related to debian packages retrieval:
+
+```bash
+docker build -f Dockerfile-tate -t tv-tate --build-arg RESOURCES=theia-app-theia-ext .
+```
+
+Once the image has been built, start a container named _tv-tate-1_ from the _tv-tate_ image:
+
+```bash
+docker run --name tv-tate-1 tv-tate
+```
+
+Find the IP address of the _tv-tate-1_ container:
+
+```bash
+docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' tv-tate-1
+```
+
+Connect to port `4000` of the IP identified earlier using your browser. You should be able to see the theia-trace-extension UI. If it is not visible, click on `View -> Open View... -> Trace Viewer`
+
+#### Connect to a trace-server
+
+Let's say you have another container running the trace-server at IP 172.17.0.2, port `8080`. You can launch the _tv-tate-1_ container and connect it to the trace-server using the following command:
+
+```bash
+docker run --name tv-tate-1 --network="host" -e TRACE_SERVER_URL=172.17.0.2:8080/tsp/api tv-tate
+```
```

Comment:
When I run this line after having run command in line 51 above I get the following error:
```
docker: Error response from daemon: Conflict. The container name "/tv-tate-1" is already in use by container "bdc7e40b3c61d60538b54e2842c34a66e9641d00b17511096d01d7db81e4782f". You have to remove (or rename) that container to be able to reuse that name.
```

It would be good to mention that the previous has to removed and provide users with the command for that for people that are not so familiar with docker.

Reply:
I have added a section at the bottom of the readme: "Docker tips and tricks", in which I gave a few ways to deal with this error, as well as a few other tips about other things. 

Since this is the first "docker run" I have also specifically added an alternative command that discards the container after the run, as an example.